src/libs/ajax/User.test.ts

```diff
@@ -1,38 +1,441 @@
-import { kvArrayToObject } from './User';
-
-// Workaround for circular import issues.
-jest.mock('src/auth/auth');
-
-describe('kvArrayToObject', () => {
-  it('converts an array of key/value objects to an object', () => {
-    // Act
-    const result = kvArrayToObject([
-      { key: 'foo', value: 1 },
-      { key: 'bar', value: 2 },
-      { key: 'baz', value: 3 },
-    ]);
-
-    // Assert
-    expect(result).toEqual({
-      foo: 1,
-      bar: 2,
-      baz: 3,
+import {
+  CreateTerraUserProfileRequest,
+  generateAPIBodyForCreateUserProfile,
+  generateAPIBodyForUpdateUserProfile,
+  OrchestrationUpsertTerraUserProfileRequest,
+  UpdateTerraUserProfileRequest,
+} from 'src/libs/ajax/User';
+import { TerraUserProfile } from 'src/libs/state';
+
+jest.mock('src/libs/ajax/ajax-common', () => ({
+  authOpts: jest.fn(() => ''),
+}));
+
+const completeUserProfile: TerraUserProfile = {
+  firstName: 'testFirstName',
+  lastName: 'testFirstName',
+  contactEmail: 'testFirstName',
```

Comment:
A more realistic mock would be nice here, especially if there's some email validation added at some point.
```suggestion
  contactEmail: 'user@example.com',
```